### ISSUE TITLE
Optimization Pass: Dropout with ratio 0.0 should be a Nop

### ISSUE THREAD

**Issue by ArmenAg** [onnx/onnx#1521]

**Comment by houseroad** [comment]
Since we only support test (inference) mode, I think in current modes... Dropout is just a nop, whether ratio is 0 or not

**Comment by ArmenAg** [comment]
@spandantiwari and I we're looking at a pytorch exported model via netron and we saw that dropout was still included in the graph. This pass would be useful to canonicalize the graph since dropout isn't actually being used.

**Comment by ArmenAg** [comment]
An okay example of this is if you have: `softmax -> dropout -> log` (i dont know why you would do this but the example works). Our logsoftmax optimization pass wouldn't work since the dropout nop is in the way. By running the dropout nop optimization pass we would get `softmax -> log`which would then be fused into `logsoftmax`. Canonicalizing the graph is really important for robust optimization.
